src/Utility/TypeExtensions.cs

```diff
@@ -141,8 +141,13 @@ public static object ConvertFromJson(string json)
 
         private static string ConvertToJson(object fromObj)
         {
+            /* we set the max-depth to 50 because the Durable Functions Extension
+             * may produce deeply nested JSON-Objects when callig its
+             * WhenAll/WhenAny APIs. The value 50 is arbitrarily chosen to be
+             * "deep enough" for the vast majority of cases.
+             */
             var context = new JsonObject.ConvertToJsonContext(
-                maxDepth: 4,
+                maxDepth: 50,
```

Comment:
This looks like a _potential_ breaking change. Please make sure you check all the ConvertToJson use cases.

Reply:
Yes, I was thinking about that. I could look to make this change _only_ for the Durable Functions bindings. Let me consider this a little longer.